File: JAVA/RomanToInteger.java
import java.util.HashMap;

class Solution {
	static HashMap<Character, Integer> map = new HashMap<Character, Integer>();
	static 
	{
        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);
	}
    public int romanToInt(String s) {
        // Start typing your Java solution below
        // DO NOT write main() function
        if(s.length() == 1)
        	return map.get(s.charAt(0));

        int sum = 0;
        int first = map.get(s.charAt(0));
        int second = 0;
        for(int i = 0; i < s.length()-1; i++)
        {
        	second = map.get(s.charAt(i+1));
        	if(first >= second)
        	{
        		sum += first;
        	}
        	else
        	{
        		sum -= first;
        	}
        	first = second;
        }
        sum += first;
        return sum;
    }
}
public class RomanToInteger {

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		// TODO Auto-generated method stub
		Solution sol = new Solution();
		System.out.println(sol.romanToInt("MMCDXVIII"));
	}

}

File: C++/RomanToInteger.cpp
#include<iostream>
#include<stdlib.h>
#include<vector>
#include<string>
#include<string.h>
#include<algorithm>
#include<math.h>
#include<map>
using namespace std;
static map<char, int> vm;
void init()
{
		vm.insert(map<char, int>::value_type('I', 1));
		vm.insert(map<char, int>::value_type('V', 5));
		vm.insert(map<char, int>::value_type('X', 10));
		vm.insert(map<char, int>::value_type('L', 50));
		vm.insert(map<char, int>::value_type('C', 100));
		vm.insert(map<char, int>::value_type('D', 500));
		vm.insert(map<char, int>::value_type('M', 1000));
}
class Solution {
public:
    int romanToInt(string s) {
		init();
		if(s.size() == 0)
			return 0;
		int sum = 0;
		int first = vm.find(s[0])->second;
		for(int i = 1; i < s.size(); i++)
		{
			int second = vm.find(s[i])->second;
			if(first >= second)
				sum += first;
			else
				sum -= first;
			first = second;
		}
		sum += first;
		return sum;
    }
};
int main()
{
	string s = "I";
	Solution sol;
	cout<<sol.romanToInt(s);
	getchar();
}